flask_rest_jsonapi/data_layers/alchemy.py

```diff
@@ -452,7 +452,12 @@ def sort_query(self, query, sort_info):
             field = sort_opt['field']
             if not hasattr(self.model, field):
                 raise InvalidSort("{} has no attribute {}".format(self.model.__name__, field))
-            query = query.order_by(getattr(getattr(self.model, field), sort_opt['order'])())
+            if sort_opt['relationship']:
+                relationField = getattr(self.model, field)
+                relationClass = relationField.mapper.class_
+                query = query.join(relationField).order_by(getattr(getattr(relationClass, 'id'), sort_opt['order'])())
```

Comment:
I think it is not good to use the id field because it can not exists. For example I never name my primary keys as id.